pinot-plugins/pinot-file-system/pinot-adls/src/test/java/org/apache/pinot/plugin/filesystem/test/ADLSGen2PinotFSTest.java

```diff
@@ -0,0 +1,395 @@
+import com.azure.storage.file.datalake.DataLakeFileSystemClient;
+import com.azure.storage.file.datalake.DataLakeServiceClient;
+import com.azure.storage.file.datalake.models.DataLakeStorageException;
+import com.azure.storage.file.datalake.models.PathItem;
+import com.azure.storage.file.datalake.models.PathProperties;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import org.apache.pinot.plugin.filesystem.ADLSGen2PinotFS;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Tests the Azure implementation of ADLSGen2PinotFS
+ */
+public class ADLSGen2PinotFSTest {
+
+  @Mock
+  private DataLakeFileSystemClient _mockFileSystemClient;
+  @Mock
+  private DataLakeDirectoryClient _mockDirectoryClient;
+  @Mock
+  private BlobServiceClient _mockBlobServiceClient;
+  @Mock
+  private BlobContainerClient _mockBlobContainerClient;
+  @Mock
+  private BlobClient _mockBlobClient;
+  @Mock
+  private BlobInputStream _mockBlobInputStream;
+  @Mock
+  private DataLakeServiceClient _mockServiceClient;
+  @Mock
+  private DataLakeFileClient _mockFileClient;
+  @Mock
+  private DataLakeStorageException _mockDataLakeStorageException;
+  @Mock
+  private SimpleResponse _mockSimpleResponse;
+  @Mock
+  private PathProperties _mockPathProperties;
+  @Mock
+  private PagedIterable _mockPagedIterable;
+  @Mock
+  private PathItem _mockPathItem;
+
+  private URI _mockURI;
+  private ADLSGen2PinotFS _adlsGen2PinotFsUnderTest;
+
+  private final static String MOCK_FILE_SYSTEM_NAME = "fileSystemName";
+
+  @BeforeMethod
+  public void setup() throws URISyntaxException {
+    MockitoAnnotations.initMocks(this);
+    _adlsGen2PinotFsUnderTest = new ADLSGen2PinotFS(_mockFileSystemClient, _mockBlobServiceClient);
+    _mockURI = new URI("mock://mock");
+  }
+
+  @AfterMethod
+  public void tearDown() {
+    verifyNoMoreInteractions(_mockDataLakeStorageException, _mockServiceClient, _mockFileSystemClient,
+        _mockSimpleResponse, _mockDirectoryClient, _mockPathItem, _mockPagedIterable, _mockPathProperties,
+        _mockFileClient, _mockBlobContainerClient, _mockBlobClient, _mockBlobServiceClient, _mockBlobInputStream);
+  }
+
+  @Test(expectedExceptions = IllegalArgumentException.class)
+  public void testInitNoAuth() {
+    PinotConfiguration pinotConfiguration = new PinotConfiguration();
+    _adlsGen2PinotFsUnderTest.init(pinotConfiguration);
+  }
+
+  @Test
+  public void testGetOrCreateClientWithFileSystemGet() {
+    when(_mockServiceClient.getFileSystemClient(MOCK_FILE_SYSTEM_NAME)).thenReturn(_mockFileSystemClient);
```

Comment:
Indentation?

Reply:
Can you elaborate ?
It is indented just like other classes in the code base.